### ISSUE TITLE
Feature request: Make npm configuration optional

### ISSUE THREAD

**Issue by iiian** [semantic-release/cli#318]
Is it a core use case or an opt-in use case that semantic-release be able to publish to npm? 

If not, I think it would increase the value of the setup tool to be able to produce configurations where npm publishing is only optional

**Comment by adrienjoly** [comment]
Yes, I don't publish all my semantically versionned repos to npm!

I would be happy to pass a flag in order to avoid being asked about my npm credentials

```sh
$ npx semantic-release-cli setup --no-npm
```

Anybody started working on something like this, by any chance?

**Comment by gr2m** [comment]
You can just add `"private": true` to the package.json

> Is it a core use case or an opt-in use case that semantic-release be able to publish to npm?

we do plan to decompose the `semantic-release` module and create `@semantic-release/core`, exactly for use cases like this where you are not interested in `npm`.
